apps/common/utils/index.tsx

```diff
@@ -83,3 +83,10 @@ export function	formatDateShort(value: number): string {
 
 	return (new Intl.DateTimeFormat([locale, 'en-US'], {year: 'numeric', month: 'short', day: '2-digit'}).format(value));
 }
+
+/* 🔵 - Yearn Finance **************************************************************************
+**	Returns an object composed of each element of an array, using one of the elements 
+**  properties as its key
+**********************************************************************************************/
+export const keyBy = <T1, T2 extends keyof T1 & string>(array: T1[], key: T2): TDict<T1 | undefined> => 
+	(array || []).reduce((r, x): TDict<T1> => ({...r, [x[key] as string]: x}), {});
```

Comment:
Looks a bit cryptic, how about having somes tests, GPT suggested the following. Probably missing some edge cases though:

```typescript
import { keyBy } from './keyBy';

type TestObject = {
  id: string;
  name: string;
};

describe('keyBy', () => {
  it('should return an empty object when the input array is empty', () => {
    const input: TestObject[] = [];
    const result = keyBy(input, 'id');
    expect(result).toEqual({});
  });

  it('should create a dictionary keyed by the specified key', () => {
    const input: TestObject[] = [
      { id: '1', name: 'Alice' },
      { id: '2', name: 'Bob' },
      { id: '3', name: 'Charlie' },
    ];
    const expected = {
      '1': { id: '1', name: 'Alice' },
      '2': { id: '2', name: 'Bob' },
      '3': { id: '3', name: 'Charlie' },
    };
    const result = keyBy(input, 'id');
    expect(result).toEqual(expected);
  });

  it('should overwrite duplicate keys with the last occurrence in the array', () => {
    const input: TestObject[] = [
      { id: '1', name: 'Alice' },
      { id: '2', name: 'Bob' },
      { id: '2', name: 'Eve' },
    ];
    const expected = {
      '1': { id: '1', name: 'Alice' },
      '2': { id: '2', name: 'Eve' },
    };
    const result = keyBy(input, 'id');
    expect(result).toEqual(expected);
  });

  it('should return an empty object when the input array is null', () => {
    const input: TestObject[] | null = null;
    const result = keyBy(input as TestObject[], 'id');
    expect(result).toEqual({});
  });

  it('should return an empty object when the input array is undefined', () => {
    const input: TestObject[] | undefined = undefined;
    const result = keyBy(input as TestObject[], 'id');
    expect(result).toEqual({});
  });
});

```